src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -61,9 +63,13 @@ public ImageTagHelper(IHostingEnvironment hostingEnvironment, IMemoryCache cache
 
         protected IMemoryCache Cache { get; }
 
+        protected IUrlHelper UrlHelper { get; }
+
         /// <inheritdoc />
         public override void Process(TagHelperContext context, TagHelperOutput output)
         {
+            Src = UrlHelper.Content(Src);
```

Comment:
Since "valid non-empty URL potentially surrounded by spaces" appears fairly frequently in the HTML 5.0 spec and is applied to all attributes updated in existing tag helpers, should at least `Trim()` the value before calling `Content()`.